File: BackEnd Spring Boot/src/main/java/com/ecommerce/entity/Product.java
package com.ecommerce.entity;


import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name="product")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String title;

    private String description;

    private int mrpPrice;

    private int sellingPrice;

    private int discountPercent;

    private int quantity;

    private String color;

//    Changes Recently
    private String brand;

    // it will create a separate table for this field
    @ElementCollection
    private List<String> images = new ArrayList<>();

    @ManyToOne
    private Category category;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;






}

File: Frontend react/src/types/product.ts
export interface Product {
  id: number;
  title: string;
  description: string;
  mrpPrice: number;
  sellingPrice: number;
  discountPercent: number;
  quantity: number;
  color: string;
  brand: string;
  images: string[];
  category: {
    id: number;
    name: string;
  };
  createdAt: string;
  updatedAt: string;
}

export interface ProductSearchParams {
  category?: string;
  brand?: string;
  colors?: string;
  minPrice?: number;
  maxPrice?: number;
  minDiscount?: number;
  sort?: string;
  pageNumber?: number;
}

export interface ProductCreateRequest {
  title: string;
  description: string;
  mrpPrice: number;
  sellingPrice: number;
  quantity: number;
  color: string;
  brand: string;
  category: string;
}

export interface ProductPage {
  content: Product[];
  totalElements: number;
  totalPages: number;
  number: number;
  size: number;
}

export interface FilterOptions {
  categories: string[];
  brands: string[];
  colors: string[];
  minPrice: number;
  maxPrice: number;
}
